src/kontrol/prove.py

```diff
@@ -316,7 +318,40 @@ def init_and_run_proof(test: FoundryTest) -> APRFailureInfo | Exception | None:
             start_server=start_server,
             port=options.port,
             maude_port=options.maude_port,
+            haskell_threads=options.max_frontier_parallel,
         ) as kcfg_explore:
+
+            def create_kcfg_explore() -> KCFGExplore:
+                if options.maude_port is None:
```

Comment:
Where does this logic come from? Is this an added feature, or something that got pushed up from `evm-semantics`?

Reply:
Yes, the `dispatch` construction is pushed up up from `evm-semantics`, from `legacy_explore`.